### ISSUE TITLE
default Electron icon is used reason=application icon is not set

### ISSUE THREAD

**Issue by Marcelh1983** [ElectronNET/Electron.NET#292]
* **ElectronNET.API 5.22.13, netcoreapp2.2, electron-builder version=20.44.4, node 10.10.0**: 

* **Target**:  windows

I use the win/icon option to add my own app icon:
```
    "win": {
      "icon": "Assets/icon.ico"
    },
```
electron-builder keeps logging: 

> default Electron icon is used reason=application icon is not set

I also tried ElectronNET.WebApp which gives the same message.

**Comment by robalexclark** [comment]
Its probably not a bug. It took me aaages to get it to work. If you don't get the path to your icon correct from the "current working folder" it defaults to the electron icon and you get that message. Take a look at this [repo ](https://github.com/robalexclark/SilveR) and the electron.manifest.json there. Determine the path to icon.png and compare to your own settings.

**Comment by Marcelh1983** [comment]
@robalexclark thank you so much! This fixed it. I created a pull request to fix it in the example aswell.

**Comment by alexwiese** [comment]
For anyone else hitting this I found that I had to prefix the directory with `bin` to get it to work:

```  
"win": {
      "icon": "bin/Assets/favicon.ico"
    },
```

**Comment by MrCircuit** [comment]
My project did not work with an ICO file. Converting it to PNG worked instead. Maybe the fact, that the ICO had two embedded resolutions broke the feature?

**Comment by GregorBiswanger** [comment]
Electron-builder is used to integrate the icon. In the electron.manifest.json file, the entire build part is the 1:1 configuration of electron-builder. Best for problems, google for electron builder and icons...

**Comment by pmagnussen** [comment]
The following worked for me:
`"win": {
      "icon": "../../../Assets/reporting_tool.ico",
      "publish": [
        {
          "provider": "github",
          "owner": "ElectronNET",
          "repo": "electron.net-api-demos",
          "token": ">> Insert GH_TOKEN here! <<"
        }
      ]
    }`

**Comment by Mondonno** [comment]
Anyone solved this? or anyone have a method how to add custom icon into the electron.net app?

**Comment by schaveyt** [comment]
@Mondonno 

~~~json
"build": {
    "appId": "...",
    . . .
    . . .
    "directories": {
      "output": "../../../bin/Desktop"
    },
    "nsis": {
      "oneClick": true,
      "perMachine": true
    },
    "win": {
       // This path is relative to the `{.net project folder}`\obj\desktop\win
      "icon": "../../../wwwroot/favicon.ico"
    },
~~~

**Comment by raydeo** [comment]
...and for mac use this.

```json
{
  "name": "YOOOOOOO!",
  "version": "1.0.0",
  "description": "the easy way to add logo and installer background",
  "main": "main.js",
  "scripts": {
    "start": "electron ."
  },
  "repository": "https://github.com",
  "keywords": [
    "Electron"
  ],
  "build": {
    "appId": "com.SUPPLIER.APP_NAME",
    "mac": {
      "category": "public.app-category.utilities",
      "target": "dmg"
    },
    "extraResources": [
      "res",
      "favicon.png"
    ],
    "directories": {
      "buildResources": "build",
      "output": "oh-yeah"
    }
  },
    ...additional stuff after


```

Make sure you got 'build' folder on same level as your package.json is located, with 'icon.png' size >= 512x512

If you want to add background splash picture to the installer just drop an image with name 'background.png' in the build folder also.


![Screenshot 2022-01-20 at 00 32 01](https://user-images.githubusercontent.com/29735453/150240703-d143c323-7a4e-4831-a2c8-3a2e93ba58a2.png)

**Comment by danatcofo** [comment]
As a side note, Mac also supports the pkg target.

**Comment by mazm786** [comment]
> For anyone else hitting this I found that I had to prefix the directory with `bin` to get it to work:
> 
> ```
> "win": {
>       "icon": "bin/Assets/favicon.ico"
>     },
> ```

Please also mention solution for linux

**Comment by Taster-git** [comment]
> Electron-builder is used to integrate the icon. In the electron.manifest.json file, the entire build part is the 1:1 configuration of electron-builder. Best for problems, google for electron builder and icons...

Please post an example of icon in Linux. Png file path is set but it displays system default icon at system tray. (Neither it is electron default nor my custom icon. While I want to set custom icon.)

**Comment by Silvenga** [comment]
FWIW - Linux has oddities documented here: https://www.electron.build/configuration/linux

Tldr: filename matters.

**Comment by ManBearPigg** [comment]
**On Linux I had to give the png dimensions as the file name.**

```
"build": {
      "appId": "com.mangearpigg.ChatGUI",
      "linux": {
        "target": "AppImage",
        "category": "Utility",
       "icon": "512x512.png"
      }
 },
```

Where the icon is just a png in the same root directory that package.json is in. To be clear, if the png is called icon.png, or icon_512x512.png, or anything else other than 512x512.png, it won't work.

**Comment by aldairl** [comment]
Config the "buildResources" folder in your config:
 "build": {
    "appId": "p",
    "productName": "p",
    "asar": true,
    "asarUnpack": "**\\*.{node,dll}",
    "files": [
      "build",
      "node_modules",
      "package.json"
    ],
    "directories": {
      "app": "./",
      "output": "app-build",
      "buildResources": "src/assets"
    },
    "win": {
      "target": [
        "nsis"
      ],
      "publish": [
        "github"
      ]
    }
  },

**Comment by Amanullah202** [comment]
Anyone knows, how to do on electron-vite (it also uses electron builder though) on linux and windows. the exported deb or win exe give same warning and use default icon. but i have icon.png in resources folder.

Also can some one tell where to add this:
 "build": {
  "icon" : "icon.png"
  },
in electron-builder.yml file or package.json file?

**Comment by aldairl** [comment]
I set the config in package.json file. look a simple config:

`...
  "devDependencies": {
    "@types/react": "^18.2.66",
      ...
  },
  "eslintConfig": {
    "extends": "standard"
  },
  "build": {
    "appId": "app_namen",
    "files": [
      "dist/**/*",
      "electron/**/*"
    ],
    "directories": {
      "buildResources": "src/assets"
    },
    "win": {
      "icon": "assets/icon.ico"
    }
  }`

**Comment by ozirisdark** [comment]
All this doesn't work (win)
